### PR TITLE
participant-state: Remove the Daml-LF transaction and value aliases. [KVL-1002]

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
They add nothing.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
looks plausible

**Review by remyhaemmerle-da — APPROVED** [review]

**Review by hubert-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/store/ActiveLedgerStateManager.scala [hunk 1]
```diff
@@ -64,7 +61,7 @@ private[platform] class ActiveLedgerStateManager[ALS <: ActiveLedgerState[ALS]](
       copy(currentState = currentState.mapAcs(f), rollbackStates = rollbackStates.map(_.mapAcs(f)))
 
     def result: Either[Set[RejectionReason], ALS] = {
-      if (!rollbackStates.isEmpty) {
+      if (rollbackStates.nonEmpty) {
         sys.error(s"IMPOSSIBLE finished transaction but rollback states is not empty")
```
Comment (miklos-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/ActiveLedgerStateManager.scala:65]:
```suggestion
        sys.error("IMPOSSIBLE finished transaction but rollback states is not empty")
```

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/PostCommitValidation.scala [hunk 2]
```diff
@@ -16,7 +17,7 @@ import com.daml.ledger.participant.state.v1.{CommittedTransaction, RejectionReas
   *
   * Post-commit validation is relevant for three reasons:
   * - keys can be referenced by two concurrent interpretations, potentially leading to
-  *   either create nodes with duplicate active keys or lookup-by-key nodes referring to
+  * either create nodes with duplicate active keys or lookup-by-key nodes referring to
```
Comment (miklos-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/PostCommitValidation.scala:20]: I'm not sure if you intended this change.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/DivulgedContract.scala [hunk 3]
```diff
@@ -16,5 +16,5 @@ import com.daml.lf.value.Value
   */
 final case class DivulgedContract(
     contractId: Value.ContractId,
-    contractInst: ContractInst,
+    contractInst: Value.ContractInst[Value.VersionedValue[Value.ContractId]],
```
Comment (miklos-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/DivulgedContract.scala:19]: We may want to move the alias for `ContractInst` to the lib where these types are being defined.

**Review by miklos-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/PostCommitValidation.scala (hunk 2 above)
Comment (SamirTalwar-DA) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/PostCommitValidation.scala:20]: Nope, thanks for pointing it out. Autoformat strikes again.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/DivulgedContract.scala (hunk 3 above)
Comment (SamirTalwar-DA) [reply ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/DivulgedContract.scala:19]: We discussed this elsewhere. The maintainers of LF feel that using the alias is a choice and they don't encourage it or discourage it, so it should be declared as and when it's needed.

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/PostCommitValidation.scala (hunk 2 above)
Comment (SamirTalwar-DA) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/PostCommitValidation.scala:20]:
```suggestion
  *   either create nodes with duplicate active keys or lookup-by-key nodes referring to
```

**Pull request merged**